flutter_appauth_platform_interface/lib/src/external_agent_type.dart

```diff
@@ -0,0 +1,14 @@
+// Enum representing the type of external agent
+// to use for the authorization flow.
+enum ExternalAgentType {
+  /// Uses ASWebAuthenticationSession, the default for iOS 12 and above.
+  asWebAuthenticationSession,
+
+  /// Uses an ephemeral session that does not share browser data with
+  /// the user's normal browser session and does not keep the cache.
+  ephemeralAsWebAuthenticationSession,
+
+  /// Uses SFSafariViewController, which does not share browser data
+  /// with the user's normal browser session but keeps the cache.
+  sfSafariViewController
```

Comment:
Would you be ok in adding a paragraph that indicates this isn't applicable to macOS. Referring to what you mentioned on how it would end up using the macOS so presumably the comment should say that on macOS it will have the same behaviour as `asWebAuthenticationSession` (?)

Reply:
Sure, added!